client/pages/matches.jsx

```diff
@@ -0,0 +1,74 @@
+import React from 'react';
+// import parseRoute from './lib/parse-route';
+
+export default class Matches extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      id: '',
+      photos: '',
+      name: '',
+      location: '',
+      age: '',
+      breed: '',
+      gender: '',
+      size: ''
+    };
+  }
+
+  componentDidMount() {
+  // call fetch on api/pets/location/type and set the state of the items in here
+  // also ensure that the catch method gets called
+    const queryString = window.location.hash.split('?');
+    const params = new URLSearchParams(queryString[1]);
+    const location = params.get('location');
+    const type = params.get('type');
+
+    fetch(`/api/pets/${location}/${type}`, {
+      method: 'GET'
+    })
+      .then(response => response.json())
+      .then(animal =>
+        this.setState({
+          id: animal[0].id,
+          photos: animal[0].primary_photo_cropped.full,
+          name: animal[0].name,
+          location: animal[0].contact.address.city,
+          age: animal[0].age,
+          breed: animal[0].breeds.primary,
+          gender: animal[0].gender,
+          size: animal[0].size
+        }))
+      .catch(error => {
+        console.error('Error', error);
+      });
+
+  }
+
+  render() {
+    // create a new variable to hold the values that we're gonna pull from the api and put them here
+    const { photos, name, location, age, breed, gender, size } = this.state;
+
+    return (
+
+      <div className="card m-5">
+        <div className="row g-0 ">
```

Comment:
Instead of showing a bunch of empty stuff, you should have a guard here and an `isLoading` state so that the user doesn't see a flash of a misshaped search result.

```js
if (this.state.isLoading) return null;
```